image/tarexport/save.go

```diff
@@ -38,6 +38,7 @@ type saveSession struct {
 	images      map[image.ID]*imageDescriptor
 	savedLayers map[string]struct{}
 	diffIDPaths map[layer.DiffID]string // cache every diffID blob to avoid duplicates
+	diffIDDescs map[layer.DiffID]distribution.Descriptor
```

Comment:
I'm kinda failing to understand why we need this map

Reply:
We need it because `saveLayer` [doesn't read and save the tar archive if it was alread saved during this session](https://github.com/moby/moby/blob/33b5a5813b805aaba28c06ee1a8aa5b1f2435a1d/image/tarexport/save.go#L485).

So in this case we need to remember the generated descriptor and return it for any subsequent calls.

Reply:
Gotcha!